src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -553,9 +558,10 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
             return BUILD_NOW;
         }
 
-        final BuildData buildData = fixNull(getBuildData(lastBuild));
-        if (buildData.lastBuild != null) {
-            listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
+        BuiltRevisionMap builtRevisions = BuiltRevisionMap.forProject(project);
```

Comment:
Doing this would load from xml any time you request BuiltRevisionMap for a project. TransientProjectActionFactory do attach to project in memory once for all